src/writer/_dirent.h

```diff
@@ -28,6 +28,55 @@ namespace zim
 {
   namespace writer {
     class Dirent;
+
+    struct TinyString {
+      TinyString() :
+        m_data(nullptr),
+        m_size(0)
+      {}
+      TinyString(const std::string& s) :
+        m_data(new char[(uint16_t)s.size()]),
+        m_size(s.size())
+      {
+        if (s.size() >= 0xFFFF) {
+          throw std::runtime_error("String len is too big");
+        }
+        std::memcpy(m_data, s.data(), m_size);
+      }
+      TinyString(TinyString&& t):
+        m_data(t.m_data),
+        m_size(t.m_size)
+      {
+        t.m_data = nullptr;
+        t.m_size = 0;
+      };
+      TinyString(const TinyString& t) = delete;
+      ~TinyString() {
+        if (m_data) {
+          delete[] m_data;
+          m_data = nullptr;
+        }
+      }
+      operator std::string() const { return std::string(m_data, m_size); }
+      bool empty() const { return m_size == 0; }
+      size_t size() const { return m_size; }
+      const char* const data() const { return m_data; }
+      bool operator==(const TinyString& other) const {
+        return (m_size == other.m_size) && (std::memcmp(m_data, other.m_data, m_size) == 0);
+      }
+      bool operator<(const TinyString& other) const {
+        auto min_size = std::min(m_size, other.m_size);
+        auto ret = std::memcmp(m_data, other.m_data, min_size);
+        if (ret == 0) {
+          return m_size < other.m_size;
+        } else {
+          return ret < 0;
+        }
+      }
+      char* m_data;
+      uint16_t m_size;
```

Comment:
Can't you avoid this field by using null-terminated strings? There can be two arguments against such an implementation:

1. Slower `size()` operation.
2. Strings containing `\0` bytes (Update: `PathTitleTinyString` matches this case).

Reply:
Yes, `PathTitleTinyString` store a `\0` bytes.
But we may remove the `m_size` attribute by using a specialized version of `size` for `PathTitleTinyString`.